src/main/java/de/fraunhofer/aisec/cpg/passes/EvaluationOrderGraphPass.java

```diff
@@ -319,15 +319,7 @@ private void handleCallExpression(@NonNull Node node) {
 
   private void handleMemberExpression(@NonNull Node node) {
     MemberExpression memberExpression = (MemberExpression) node;
-    // analyze the base
-    if (memberExpression.getBase() instanceof Statement) {
-      createEOG(memberExpression.getBase());
-    }
-
-    // analyze the member
-    if (memberExpression.getMember() instanceof Statement) {
-      createEOG(memberExpression.getMember());
-    }
+    // TODO how to handle base?
```

Comment:
Do we explicitly create the EOG to the refers or or is already done by handleDeclaredReferenceExpression? Otherwise we need to create EOG to refers and to base.

Reply:
If I understand the mechanics correctly, the handleDeclaredReferenceExpression won't be called if there is a handler for MemberExpression. So we'll have to handle all parts of the EOG here. @konradweiss what would be the optimal EOG pattern for MemberExpressions?

Reply:
@Masrepus As the member in theory can contain an Expression, although it was pointed out by @oxisto that not all expression are allowed, the member and the base need to be evaluated and therefore an edge is needed to both, the order can be found in the old code as far as I known. handleMemberExpression should be called first by the handler, as it is more specific than handleDeclaredReferenceExpression. Does this answer the question?